compiler/rustc_hir_analysis/src/collect/resolve_bound_vars.rs

```diff
@@ -717,6 +717,21 @@ impl<'a, 'tcx> Visitor<'tcx> for BoundVarContext<'a, 'tcx> {
         }
     }
 
+    #[instrument(level = "debug", skip(self))]
+    fn visit_pattern_type_pattern(&mut self, p: &'tcx hir::Pat<'tcx>) {
+        if let hir::PatKind::Range(start, end, _) = p.kind {
+            if let Some(start) = start {
+                self.visit_expr(start);
+            }
+
+            if let Some(end) = end {
+                self.visit_expr(end);
+            }
+        } else {
+            intravisit::walk_pat(self, p)
```

Comment:
Doesn't walk_pat already visit the nested expressions?

Reply:
Right, done!